tests/integration_tests/functional/test_vsock.py

```diff
@@ -179,3 +181,106 @@ def test_vsock_epipe(
     # If this ever fails due to 100 closes before read() we must
     # add extra tooling that will trigger only writes().
     assert metrics['signals']['sigpipe'] > 0
+
+
+def test_vsock_transport_reset(
+        test_microvm_with_ssh,
+        network_config,
+        bin_vsock_path,
+        test_fc_session_root_path
+):
+    """
+    Vsock transport reset test.
+
+    Steps:
+    1. Start echo server on the guest
+    2. Start host workers that ping-pong data between guest and host,
+    without closing any of them
+    3. Pause VM -> Create snapshot -> Resume VM
```

Comment:
We should also test the scenario when a VM is created with LoadSnapshot.

Reply:
Will do.